src/manifest.rs

```diff
@@ -94,6 +94,33 @@ impl Manifest {
 
         sections
     }
+
+    /// returns features exposed by this manifest
+    pub fn features(&self) -> Vec<String> {
+        match self.data.as_table().get("features") {
+            None => vec![],
+            Some(item) => match item {
+                toml_edit::Item::None => vec![],
+                toml_edit::Item::Value(v) => vec![v.to_string()],
+                toml_edit::Item::Table(t) => t
+                    .get_values()
+                    .iter()
+                    .map(|(keys, _val)| keys.iter().map(|k| k.to_string()))
+                    .flatten()
+                    .collect(),
+                toml_edit::Item::ArrayOfTables(a) => a
```

Comment:
I'm used to features being a table where each key points to a list of strings.

It can also be a string and an array of tables?  Where can I read more about this?

Reply:
so, the reason that I handle string and array of table cases is more because of the fact that `toml_edit::Item` has those enum variants. I did look at the manifest spec: https://doc.rust-lang.org/cargo/reference/features.html and it does mention off hand that `features` should be a table which would mean that the value or array of tables is not allowed, but I haven't actually tested that. 

I think it would be equivalent at the moment to unify the `value` and `ArrayOfTables` into a `_ => unreachable!()` but the logic isn't much more complex and it is a bit of a grey area, so unless someone is certain that only tables are allowed here we could remove it, I'd vote for just leaving it in anyway in case it is actually valid behaviour. I'm not very set on this idea though, if you think removing is still better, I'm okay with that.

Reply:
> , so unless someone is certain that only tables are allowed here we could remove it,

I'd go the opposite, let's start with tables and add more as they become know.  We don't know what other formats even look like to know how to support them, if they exist.

For these cases, we should either return `vec![]` or  an error.

Reply:
noted, I'll add that change